packages/react-router/src/Matches.tsx

```diff
@@ -41,10 +42,12 @@ declare module '@tanstack/router-core' {
 
 export function Matches() {
   const router = useRouter()
+  const rootRoute: AnyRoute = router.routesById['__root__']
```

Comment:
do we need to be defensive here and type this as `AnyRoute | undefined`. I think a `__root__` route must always exist ...

Reply:
please use `rootRouteId` from `@tanstack/router-core` instead of `__root__`